Develop rebase with main branch

baziorek opened the pull request:
In iroha-python library many newest merges was being merged into `main` branch instead of develop, that is why many changes are not available on `develop`.
So I've made:
```
git cherry-pick 41a724b3f3f21921ad455d39fb959f45c98fe986
```
to fetch: https://github.com/hyperledger/iroha-python/commit/41a724b3f3f21921ad455d39fb959f45c98fe986

it will contain commits which change `iroha/iroha.py` file:

1. https://github.com/hyperledger/iroha-python/commit/41a724b3f3f21921ad455d39fb959f45c98fe986
2. https://github.com/hyperledger/iroha-python/commit/92d81772ab5c59b5521f0475300ebbc1e97f5974
3. https://github.com/hyperledger/iroha-python/commit/4d10eb0fb64bb033c950a8b6a63c6fae25fe4373
4. https://github.com/hyperledger/iroha-python/commit/3968c3903c1b7c1c2c1f2519699424416cbbe248

Rest of commits are not cherry-picked (connected with documentation, Jenkins, etc)

baziorek commented:
@Pawlak00 could You please take a look if I haven't damaged any Your changes?

Pawlak00 commented on iroha/iroha.py:
```diff
@@ -197,7 +255,8 @@ def query(self, name, counter=1, creator_account=None,
               last_tx_time=None, first_tx_height=None,
-              last_tx_height=None, **kwargs):
+              last_tx_height=None, ordering_sequence=None,
+              **kwargs):
```
ordering sequence should be specified before first_tx_time I think (to make it same as in other libraries and Iroha)

baziorek replied:
You are right, thanks.

Pawlak00 replied:
Before first_Tx_time and after first_tx_hash: https://iroha.readthedocs.io/en/develop/develop/api/queries.html

baziorek replied:
Thanks, corrected in https://github.com/hyperledger/iroha-python/pull/83/commits/943998e2a90bd48d5378e9ec4d9ea2d9fe75cebc

Resolved

baziorek commented:
@LiraLemur @iceseer I've just added more things to the PR:
-----
1. changes in protobuf files - it was done with attached script `scripts/download-schema.py`
2. updated auto-generated protobuf files - it was done by script `scripts/compile-proto.py` - **You can ignore that files in review** (`iroha/*_pb2.py`)
3. Two examples: `examples/tls-example.py` and `examples/tx-example.py`
4. Others:
    1. `setup.py` and `dev-requirements.txt` - updated versions of dependencies
    2. `SECURITY.md` - copied from `main` branch

baziorek commented:
@iceseer Thanks for approve! So I'm merging its:).
